services/frontend/domain/usecases/process_manifest_files.go

```diff
@@ -14,152 +10,64 @@ import (
-		}(manifestFile)
+	// Define config
+	config := &pb.Config{
+		Name:             unmarshalledManifest.Name,
+		ManifestFileName: fmt.Sprintf("secret_%s.file_%s", rawManifest.SecretName, rawManifest.FileName),
+		Manifest:         string(rawManifest.Manifest),
 	}
 
-	go func() {
-		waitGroup.Wait()
-		close(manifestProcessingResultsChan)
-	}()
-
-	// Collect processing results of manifest files which were processed successfully
-	for manifestProcessingResult := range manifestProcessingResultsChan {
-		var manifestName string
-		var isConfigRemoved bool
-
-		// Remove the config from configs slice.
-		// After the for loop finishes, the configs variable will contain only those configs which represent
-		// deleted manifest files -> configs which needs to be deleted.
-		configs, isConfigRemoved = removeConfig(configs, manifestProcessingResult.manifestFilepath)
-		// Check for the error first, before referencing any variables.
-		if manifestProcessingResult.processingError != nil {
-			log.Error().Msgf("Skipping over file %v due to processing error : %v", manifestProcessingResult.manifestFilepath, manifestProcessingResult.processingError)
-			continue
-		}
-		manifestName = manifestProcessingResult.unmarshalledManifest.Name
-
-		config := &pb.Config{
-			Name:             manifestName,
-			ManifestFileName: manifestProcessingResult.manifestFilepath,
-			Manifest:         string(manifestProcessingResult.rawManifestData),
-		}
-
-		err = u.ContextBox.SaveConfig(config)
-		if err != nil {
-			log.Error().Msgf("Failed to save config %v due to error : %v", manifestName, err)
-			continue
-		}
-
-		log.Info().Msgf("Details of the manifest file %s has been saved to context-box database", manifestProcessingResult.manifestFilepath)
-
-		// if the config is not in the context-box DB we start to track it.
-		if !isConfigRemoved {
-			for _, k8sCluster := range manifestProcessingResult.unmarshalledManifest.Kubernetes.Clusters {
-				if _, ok := u.inProgress.Load(k8sCluster.Name); !ok {
-					u.inProgress.Store(k8sCluster.Name, config)
-				}
-			}
-		}
+	if err := u.ContextBox.SaveConfig(config); err != nil {
+		log.Err(err).Msgf("Failed to save config %v due to error. Skipping...", unmarshalledManifest.Name)
+		return
 	}
+	log.Info().Msgf("Created config for input manifest %s", unmarshalledManifest.Name)
+}
 
-	// The configs variable now contains only those configs which represent deleted manifests.
-	// Loop over each config and request the context-box microservice to delete the config from its database as well.
-	for _, config := range configs {
-		if _, isConfigBeingDeleted := u.configsBeingDeleted.Load(config.Id); isConfigBeingDeleted {
-			continue
-		}
-		u.configsBeingDeleted.Store(config.Id, nil)
-
-		for _, k8sCluster := range config.GetCurrentState().GetClusters() {
-			if _, ok := u.inProgress.Load(k8sCluster.ClusterInfo.Name); !ok {
-				u.inProgress.Store(k8sCluster.ClusterInfo.Name, config)
-			}
-		}
```

Comment:
Could we also add this to the `deleteConfig` func ?

This code was there so that the frontend starts tracking the deletion when it got picked up.
This is so if by any chance the deletion fails immediately after the frontend service picks it up (for example builder is in a crashloop) it will be then printed in the logs that the deletion failed

Since we work with the manifest this would look like.
```go
                for _, k8sCluster := range unmarshalledManifest.Kubernetes.Clusters {
			if _, ok := u.inProgress.Load(k8sCluster.ClusterInfo.Name); !ok {
				u.inProgress.Store(k8sCluster.ClusterInfo.Name, config)
			}
		}
```